pytype/pytd/typeshed.py

```diff
@@ -144,32 +144,41 @@ def _load_third_party_packages(self):
     stubs/ contains type information for third-party packages. Each top-level
     directory corresponds to one PyPI package and contains one or more modules,
     plus a metadata file (METADATA.toml). If there are separate Python 2 stubs,
-    they live in an @python2 subdirectory. METADATA.toml
-    takes @python2 into account, so if a package has both foo.pyi and
-    @python2/foo.pyi, METADATA.toml will contain `python2 = True`.
+    they live in an @python2 subdirectory. If stubs support both Python 2 and
+    Python 3 without a separate @python2 directory, METADATA.toml will contain
+    `python2 = True`.
 
     Returns:
       A mapping from module name to a set of
       (package name, major_python_version) tuples.
     """
     metadata = {}
     modules = collections.defaultdict(set)
+    py2 = False  # does a @python2 directory exist
```

Comment:
Ah, not sure how I didn't notice this before, but I think py2 and py3 need to be defined inside the loop and stored in the 'modules' dict. (The reason this method is structured in this way - one loop over all the files in "stubs" and another to populate 'packages' - is that due to how we run pytype internally, file operations are expensive, so we want to do as few of them as possible.)

Sorry for the extra round trip.

Reply:
Whoops. Not sure how I missed this. Will fix.